### ISSUE TITLE
False negative for Lint/Void for guard clause with forgotten `return`

### ISSUE THREAD

**Issue by owst** [rubocop/rubocop#12979]
## Expected behavior

`Lint/Void` should find an offense due to the literal `true` in a void context. 

## Actual behavior

No offense found.

## Steps to reproduce the problem

```ruby
def foo
  # Intended to be `return true unless ...` but the `return` was forgotten - expected offense here
  true unless Feature.enabled?(:perform_side_effects)

  some_side_effect
end
```

## RuboCop version

```
1.64.1 (using Parser 3.3.2.0, rubocop-ast 1.31.3, running on ruby 3.3.1) [x86_64-darwin22]
```

**Comment by owst** [comment]
Wow - a super fast fix! Thanks @koic 👍
